### DISCUSSION TITLE
Data type added by extension not available in table editor

### DISCUSSION THREAD

**Discussion by exotfboy** [supabase/supabase#8702] [Questions] [answered]
When I enable the extension `postgis`  in `extension` schema and re-connect to the project, then I tried to create a new table, however when for a given column I can not find the `geometry` data type in the `type` select. Also I have checked to make sure that the schema `extension` is in the `search_path`.

And I tried to connect to db by heidisql and edit the table by adding a column, I can select the `geometry` data type. Everything show that the extension is enable successfully.

However when I re-set the `postgis` extension in `public` schema, I can find the `geometry` type in the `type select` when edit a table/column.

Is this expected?

**Comment by GaryAustin1** [comment, marked as answer]
In general you need to add/manipulate custom and extension data types with the SQL editor or an external database tool.  
 Once added they show up like this:
 
![image](https://user-images.githubusercontent.com/54564956/187340988-93b9d0be-399d-43bc-b6b1-a69e2fa04fb7.png)

Your last comment is odd, I've not seen anyone mention it working with the extension the public schema, but maybe no one has tried, or this is a new feature.  Or I've just missed comments on that "working", which is possible.

**Reply by GaryAustin1** [reply]
You should follow up in the thread, not in a new answer....

The reason that shows up for me is that I added it for my location column using the SQL editor, or Datagrip (my third party tool), don't remember which.

Yes, you would alter the table/column in the SQL editor.

**Comment by exotfboy** [comment]
When I say "geometry data type is not avaible in the Type select" I mean I can not see the `geometry` at all, not to speak of the `Custom non-......` in the above image.

> you need to add/manipulate custom and extension data types with the SQL editor or an external database tool
Do you mean that I have to alter a column from the sql editor?
